tests/admin/apple-actions/index/test-class-export.php

```diff
@@ -91,5 +113,105 @@ public function testBylineFormatWithHashtag() {
+			'apple_news_sections',
+			array(
+				(object) array(
+					'createdAt' => '2017-01-01T00:00:00Z',
+					'id' => 'abcdef01-2345-6789-abcd-ef012356789a',
+					'isDefault' => true,
+					'links' => (object) array(
+						'channel' => 'https://u48r14.digitalhub.com/channels/abcdef01-2345-6789-abcd-ef0123567890',
+						'self' => $self,
+					),
+					'modifiedAt' => '2017-01-01T00:00:00Z',
+					'name' => 'Main',
+					'shareUrl' => 'https://apple.news/AbCdEfGhIj-KlMnOpQrStUv',
+					'type' => 'section',
+				),
+			)
+		);
+
+		// Get sections for the post
+		$sections = \Admin_Apple_Sections::get_sections_for_post( $post_id );
+
+		// Check that the correct mapping was returned
+		$this->assertEquals(
+			$sections,
+			array( $self )
+		);
+
+		// Remove the transient and the map
+		delete_option( \Admin_Apple_Sections::TAXONOMY_MAPPING_KEY );
+		delete_transient( 'apple_news_sections' );
+	}
+
+	public function testThemeMapping() {
+		$themes = new \Admin_Apple_Themes();
+		$defaults = $this->getFormattingSettings( $this->settings->all() );
+		update_option( $themes->theme_key_from_name( 'Default' ), $defaults );
+
+		// Make the settings different for this theme to differentiate
+		$new_theme = $defaults;
+		$new_theme['body_color'] = '#123456';
+		update_option( $themes->theme_key_from_name( 'Test Theme' ), $new_theme );
+
+		// Create a post
+		$title = 'My Title';
+		$content = '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras tristique quis justo sit amet eleifend. Praesent id metus semper, fermentum nibh at, malesuada enim. Mauris eget faucibus lectus. Vivamus iaculis eget urna non porttitor. Donec in dignissim neque. Vivamus ut ornare magna. Nulla eros nisi, maximus nec neque at, condimentum lobortis leo. Fusce in augue arcu. Curabitur lacus elit, venenatis a laoreet sit amet, imperdiet ac lorem. Curabitur sed leo sed ligula tempor feugiat. Cras in tellus et elit volutpat.</p>';
+
+		$post_id = $this->factory->post->create( array(
+			'post_title' => $title,
+			'post_content' => $content,
+		) );
+
+		// Set it to a fake section
+		$section_id = 'https://u48r14.digitalhub.com/channels/abcdef01-2345-6789-abcd-ef012356789a';
+		update_post_meta( $post_id, 'apple_news_sections', array( $section_id ) );
+
+		// Create a mapping from that section to the test theme
+		update_option( \Admin_Apple_Sections::THEME_MAPPING_KEY, array(
+			basename( $section_id ) => 'Test Theme',
+		) );
+
+		$sections = \Admin_Apple_Sections::get_sections_for_post( $post_id );
+		$export = new Export( $this->settings, $post_id, $sections );
+		$exporter = $export->fetch_exporter();
+		$exporter->generate();
+		$json = $exporter->get_json();
+		$settings = json_decode( $json );
+
+		$this->assertEquals( $settings->componentTextStyles->dropcapBodyStyle->textColor, $new_theme['body_color'] );
+
+		// Clean up
+		delete_option( $themes->theme_key_from_name( 'Default' ) );
+		delete_option( $themes->theme_key_from_name( 'Test Theme' ) );
```

Comment:
Do you also need to `delete_option( \Admin_Apple_Sections::THEME_MAPPING_KEY )` ?